lib/spack/spack/bootstrap.py

```diff
@@ -226,6 +228,23 @@ def ensure_bootstrap_configuration():
                         yield
 
 
+def store_path():
+    """Path to the store used for bootstrapped software"""
+    enabled = spack.config.get('bootstrap:enable', True)
+    if not enabled:
+        msg = ('bootstrapping is currently disabled. '
+               'Modify config.yaml to enable it')
+        raise RuntimeError(msg)
+
+    bootstrap_root_path = spack.config.get(
+        'bootstrap:root', None
+    ) or spack.paths.user_bootstrap_path
```

Comment:
Fixed in 69a2487

Reply:
The "or" idiom is necessary to be permissive and allow explicit "null" values for "root". See 206904d Let me know if this needs further thinking.

Reply:
Shouldn't the validation require a string, in which case the value can't be explicitly `None` anyway?

Reply:
Maybe it's overly defensive, since we ship a sensible default, but the rationale here was to protect against either the empty string or the config file being deleted. I changed the implementation to be a bit less defensive but simpler in 783f9b7